tokenizer.go

```diff
@@ -54,27 +54,139 @@ func (t *Tokenizer) Close() error {
 	return nil
 }
 
+type Encoding struct {
+	IDs               []uint32
+	TypeIDs           []uint32
+	SpecialTokensMask []uint32
+	AttentionMask     []uint32
+	Tokens            []string
+}
+
+type EncodeOptions struct {
+	AddSpecialTokens C.bool
+
+	ReturnTypeIDs           C.bool
+	ReturnTokens            C.bool
+	ReturnSpecialTokensMask C.bool
+	ReturnAttentionMask     C.bool
+}
+
+type EncodeOption func(eo *EncodeOptions)
+
+func uintVecToSlice(arrPtr *C.uint, len int) []uint32 {
+	arr := unsafe.Slice(arrPtr, len)
+	slice := make([]uint32, len)
+	for i, v := range arr {
+		slice[i] = uint32(v)
+	}
+	return slice
+}
+
+func CBool(b bool) C.bool {
+	return C.bool(b)
+}
```

Comment:
i feel this doesn't deserve to be a func

Reply:
ok, I will remove it.

Reply:
done